app/api/_services/user.ts

```diff
@@ -25,11 +26,17 @@ interface UserProps extends Omit<IUser, "_id" | "profileId" | "rolesId"> {
  * @returns A boolean indicating whether the user is valid or not.
  */
 export const validateUser = async (email: string, password: string) => {
-  const user = await getUser({ email }, ["email", "password"]);
-  if (!user) {
-    return false;
+  try {
+    const user = await getUser({ email }, ["email", "password"]);
+
+    if (!user || !bcrypt.compareSync(password, user.password)) {
+      return false;
+    }
+
+    return await user.toObject();
+  } catch (e) {
```

Comment:
Why return the user object instead of a boolean?

Reply:
The user data needs to be retained for subsequent use in the JWT signing process. See: https://github.com/INEEDAMONITOR/Role-Rally/blob/main/app/api/login/route.ts